File: 3201-distribute-candies-among-children-ii/3201-distribute-candies-among-children-ii.java
class Solution {
    public long distributeCandies(int n, int limit) {
        if (n > 3 * limit) return 0;
        limit = Math.min(limit, n);
        long result = 0;
        for (int i = 0; i <= limit; i++) {
            int candy = n - i;
            int min = Math.max(0, candy - limit);
            int max = Math.min(limit, candy);
            result += Math.max(0, max - min + 1);
        }
        return result;
    }
}

File: 3201-distribute-candies-among-children-ii/3201-distribute-candies-among-children-ii.cpp
class Solution {
public:
    long long distributeCandies(int n, int limit) {
        long long res = 0;
        for (int i = 0; i <= min(n, limit); i++) {
            int rem = n - i;
            int mini = max(0, rem - limit);
            int maxi = min(limit, rem);
            res += max(0, maxi - mini + 1);
        }
        return res;
    }
};